core-js/src/main/javascript/cdf-legacy/components/input.js

```diff
@@ -525,40 +525,72 @@ var SelectMultiComponent = SelectBaseComponent.extend({
 var TextInputComponent = BaseComponent.extend({
   update: function() {
     var myself = this;
-    var name = myself.name;
-    var selectHTML = "<input type='text' id='" + name + "' name='"  + name +
-      "' value='" + Dashboards.getParameterValue(myself.parameter) +
-      (myself.size ? ("' size='" + myself.size) : (myself.charWidth ? ("' size='" + myself.charWidth) : "" ) ) +
-      (myself.maxLength ? ("' maxlength='" + myself.maxLength) : (myself.maxChars ? ("' maxlength='" + myself.maxChars) : "" ) ) + "'>";
-    if(myself.size) {
-      Dashboards.log("Warning: attribute 'size' is deprecated");
-    }
-    if(myself.maxLength) {
-      Dashboards.log("Warning: attribute 'maxLength' is deprecated");
-    }
 
-    myself.placeholder().html(selectHTML);
+    myself._addHtmlToPlaceholder();
 
-    var el = $("#" + name);
+    var el = $("#" + myself.name);
 
-    el
-      .change(function() {
-        if(Dashboards.getParameterValue(myself.parameter) !== el.val()) {
-          Dashboards.processChange(name);
-        }
-      })
-      .keyup(function(ev) {
-        if(ev.keyCode == 13 &&
-          Dashboards.getParameterValue(myself.parameter) !== el.val()) {
+    el.change(function() {
+      if (myself._isValueChanged(el)) {
+        myself._runValidation(el.val());
+        Dashboards.processChange(myself.name);
+      }
+    }).keyup(function(ev) {
+      if ((myself.refreshOnEveryKeyUp || ev.keyCode === 13) && myself._isValueChanged(el)) {
+        myself._runValidation(el.val());
+        Dashboards.processChange(myself.name);
+      }
+    });
 
-          Dashboards.processChange(name);
-        }
+    if (myself.clearButton) {
+      $("#" + myself.name + "-clear-icon").click(function () {
+        el.val("");
+        Dashboards.processChange(myself.name);
```

Comment:
Yes it makes perfect sense, it needs to be done to solve the issue of removing the red border when clearing the value